### PR TITLE
chore: upgrade ts-results-es

### PR DESCRIPTION
**Pull request by jacoobes**
chore
- upgrade ts-results-es to 4.0.0
- update context and corresponding things to it (adjusted to ts-results-es new API)
- removed check for @sern/client

### CODE REVIEW

File: src/core/structures/context.ts [hunk 1]
```diff
@@ -31,53 +31,73 @@ export class Context extends CoreContext<Message, ChatInputCommandInteraction> {
     }
 
     public get id(): Snowflake {
-        return this.ctx.val.id;
+        return safeUnwrap(this.ctx
+                            .map(m => m.id)
+                            .mapErr(i => i.id));
     }
 
     public get channel() {
```
Comment (Murtatrxx) [inline src/core/structures/context.ts]: Why this isn't typed while almost every other one does

File: test/handlers/id.test.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@ function createRandomCommandModules() {
         CommandType.Button,
     ];
     return commandModule({
-        type: randomCommandType[Math.floor(Math.random() * randomCommandType.length)],
+        type: faker.helpers.uniqueArray(randomCommandType, 1)[0],
         description: faker.string.alpha(),
         name: faker.string.alpha(),
         execute: () => {},
```
Comment (Murtatrxx) [inline test/handlers/id.test.ts:24]: Do we really need faker only for this?

Comment (Murtatrxx) [reply test/handlers/id.test.ts:24]: Oh it's now even new

Comment (Murtatrxx) [reply test/handlers/id.test.ts:24]: Nvm it's keep okay to keep

**Review by Murtatrxx — APPROVED** [review]

**Review by Murtatrxx — APPROVED** [review]

File: src/core/structures/context.ts (hunk 1 above)
Comment (jacoobes) [reply src/core/structures/context.ts:39]: it is inferred. Im not touching the return types of context until we decide what to do with sern cord

**Pull request merged**